x/chainmanager/keeper/genesis.go

```diff
@@ -5,13 +5,19 @@ import (
 	"fmt"
 
 	"github.com/0xPolygon/heimdall-v2/x/chainmanager/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 // InitGenesis sets chainmanager information for genesis.
 func (k Keeper) InitGenesis(ctx context.Context, data *types.GenesisState) {
 	if err := k.SetParams(ctx, data.Params); err != nil {
 		panic(fmt.Errorf("failed to set chainmanager params: %w", err))
 	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
```

Comment:
Hmmm.... if the migration is already done, are we sure this code will be hit?

The only way I see this code could ever be hit, is if a node was started from scratch, and was made to block-synch all the way to the tip (still feasible as the chain is only a few weeks old).

HOWEVER: Bear in mind that you're setting a value in the avl DB. So if a node starting from scratch does this, but the actual nodes that participated in the chain migration last month didn't do it, the node will halt with an app mismatch on the first block it tries to apply.
Have you tested this in a devnet/testnet?